apps/examples/st_things/sample_light_things.c

```diff
@@ -0,0 +1,135 @@
+#include <signal.h>
+#include <stdbool.h>
+#include <st_things/st_things.h>
+#include <iotbus/iotbus_gpio.h>
+
+#define TAG		"LIGHT_THINGS"
+#define SW703_BUTTON	44
+
+static const char *RES_SWITCH = "/switch";
+
+extern bool handle_get_request_on_switch(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep);
+
+static void gpio_callback_event(void *user_data)
+{
+	printf("gpio_callback_event!!\n");
+	printf("reset :: %d\n", st_things_reset());
+}
+
+static void handle_reset_result(bool result)
+{
+	printf("[%s]Reset %s.\n", result ? "succeeded" : "failed", TAG);
+}
+
+static void handle_things_status_change(st_things_status_e things_status)
+{
+	printf("[%s]Things status is changed: %d\n", TAG, things_status);
+}
+
+static void handle_main_loop(void)
+{
+//TODO : easysetup terminator
+	while (1) { 
+		sleep(1);
+	}
+}
+
+static bool handle_reset_request(void)
+{
+	printf("[%s]Received a request for RESET.\n", TAG);
+	return true;
+}
+
+static bool handle_ownership_transfer_request(void)
+{
+	printf("[%s]Received a request for Ownership-transfer. \n", TAG);
+	return true;
+}
+
+static bool handle_get_request(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	printf("Received a GET request on %s\n", req_msg->resource_uri);
+
+	bool ret = false;
+
+	if (0 == strcmp(req_msg->resource_uri, RES_SWITCH)) {
+		ret = handle_get_request_on_switch(req_msg, resp_rep);
+	} else {
+		printf("Not supported uri.\n");
+	}
+
+	return ret;
+}
+
+static bool handle_set_request(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	printf("Received a SET request on %s\n", req_msg->resource_uri);
+
+	if (0 == strcmp(req_msg->resource_uri, RES_SWITCH)) {
+		handle_set_request_on_switch(req_msg, resp_rep);
+	} else {
+		return false;
+	}
+
+	return true;
+}
+
+int ess_process(int argc, char *argv[])
+{
+	bool easysetup_complete = false;
+	st_things_initialize("/rom/sampleDevice.json", &easysetup_complete);
+	iotapi_initialize();
+
+	iotbus_gpio_context_h m_gpio = iotbus_gpio_open(SW703_BUTTON);
+
+	if (!easysetup_complete) {
+		// st_things_deinitialize();
+		// return 0;
+	}
+
+	st_things_register_request_cb(handle_get_request, handle_set_request);
+	st_things_register_reset_cb(handle_reset_request, handle_reset_result);
+	st_things_register_user_confirm_cb(handle_ownership_transfer_request);
+	st_things_register_things_status_change_cb(handle_things_status_change);
+
+	iotbus_gpio_register_cb(m_gpio, IOTBUS_GPIO_EDGE_RISING, gpio_callback_event, (void *)m_gpio);
+
+	st_things_start();
+
+	printf("[%s]=====================================================\n", TAG);
+	printf("[%s]                    Stack Started                    \n", TAG);
+	printf("[%s]=====================================================\n", TAG);
+
+	handle_main_loop();
```

Comment:
Inside handle_main_loop, there is no exit case. How can we stop this example? Isn't there any issue on SVACE about unreachable codes?

Reply:
Most of users do not want to stop Easysetup. Also we don't know exact timing when this loop should be stopped.
I did not know about SVACE.

Reply:
modified